### PR TITLE
Add Depth First Traversal [python]

### LINKED ISSUE #558: Depth First Traversal [Python]
#### This issue is about [Depth First Traversal](https://en.wikipedia.org/wiki/Depth-first_search).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
	- [x] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by amineHorseman**
Fixes #558

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

**Comment by singhpratyush** [comment]
Hi! Thanks for showing interest in contributing to this repository.

Please make sure that you have ticked the points mentioned in PR description correctly.

Thanks again!

File: depth_first_traversal/DepthFirstTraversal.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+def depthFirstTraversal(graph, node, visited):
```
Comment (singhpratyush) [inline depth_first_traversal/DepthFirstTraversal.py]: Great code. Thanks for that!

Would you like to add the iterative approach too?

Comment (amineHorseman) [reply depth_first_traversal/DepthFirstTraversal.py]: Sure, I'll add it in the same file

File: depth_first_traversal/DepthFirstTraversal.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+def main():
+    graph = {'A': ['B', 'C', 'E'],
+             'B': ['D', 'F'],
+             'C': ['G'],
+             'D': [],
+             'E': ['F'],
+             'F': ['C'],
+             'G': []}
+
+    print "Recusrive:"
```
Comment (singhpratyush) [inline depth_first_traversal/DepthFirstTraversal.py]: Please use parentheses with `print`. The code needs to be `Python3` compatible too.

File: depth_first_traversal/DepthFirstTraversal.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+             'E': ['F'],
+             'F': ['C'],
+             'G': []}
+
+    print("Recusrive:")
+    result = depthFirstTraversalRecursive(graph, 'A', [])
+    for node in result:
+        print(node)
+
+    print "Iterative:"
```
Comment (singhpratyush) [inline depth_first_traversal/DepthFirstTraversal.py]: This need to be fixed too.

Comment (amineHorseman) [reply depth_first_traversal/DepthFirstTraversal.py]: yes, sorry

**Pull request merged**